service/controller/v2/resource_set.go

```diff
@@ -146,17 +137,12 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 	}
 
 	handlesFunc := func(obj interface{}) bool {
-		customObject, err := key.ToCustomObject(obj)
+		cr, err := key.ToCustomObject(obj)
 		if err != nil {
 			return false
 		}
 
-		if key.VersionBundleVersion(customObject) == VersionBundle().Version {
-			return true
-		}
-		// TODO remove this hack with the next version bundle version or as soon as
-		// all certconfigs obtain a real version bundle version.
```

Comment:
This should be fine now. I checked a cluster that is over a year old and the cert configs have all the version set.